client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -84,38 +86,27 @@ const announceStatusChangeFailure = ( { dispatch }, action ) => {
 	);
 };
 
-export const requestComment = ( store, action ) => {
+export const requestComment = action => {
 	const { siteId, commentId, query } = action;
-	store.dispatch(
-		http( {
-			method: 'GET',
-			path: `/sites/${ siteId }/comments/${ commentId }`,
-			apiVersion: '1.1',
-			query,
-			onSuccess: action,
-			onFailure: action,
-		} )
-	);
+	return http( {
+		method: 'GET',
+		path: `/sites/${ siteId }/comments/${ commentId }`,
+		apiVersion: '1.1',
+		query,
+		onSuccess: action,
```

Comment:
As `onSuccess` and `onFailure` are the same, can we just pass the `action` as the second argument to `http()`?

Reply:
trying this now

Reply:
its weird that it also sets the onProgress mark :/